internal/tiltfile/tiltfile_state.go

```diff
@@ -591,10 +592,33 @@ func (s *tiltfileState) assembleK8sV1() error {
 
 }
 
-type claimPair struct {
-	resource       *k8sResource
-	selector       k8sObjectSelector
-	selectorString string
+func entitiesToFullName(entities []k8s.K8sEntity) (map[string]k8s.K8sEntity, error) {
+	ret := make(map[string]k8s.K8sEntity, len(entities))
+
+	for _, e := range entities {
+		// TODO(dmiller): if it's already there error
+		ret[entityToFullName(e)] = e
+	}
+
+	return ret, nil
+}
+
+func entityToFullName(e k8s.K8sEntity) string {
```

Comment:
i'm surprised that this code doesn't already exist in the k8s package, or at least in `tiltfile/k8s.go` (cf `k8sObjectID.String()`), doesn't necessarily need to change now but i'm a little antsy about similar logic being reimplemented in multiple different places.

Reply:
Fullname is a concept that we made up. Since it isn't used in other places yet (but will in the future) let's leave it here. I am pretty confident that it will move somewhere else in a subsequent PR though.